fuse/test/cachecontrol_test.go

```diff
@@ -95,6 +95,39 @@ func TestCacheControl(t *testing.T) {
 		}
 	}
 
+	// assertCacheRead asserts that file's kernel cache is retrieved as dataOK.
+	assertCacheRead := func(subj, dataOK string) {
+		t.Helper()
+
+		assertCacheReadAt := func(offset int64, size int, dataOK string) {
+			t.Helper()
+			buf := make([]byte, size)
+			n, st := fsconn.FileRetrieveCache(file.Inode(), offset, buf)
+			if st != fuse.OK {
+				t.Fatalf("%s: retrieve cache @%d [%d]: %s", subj, offset, size, st)
+			}
+			if got := buf[:n]; string(got) != dataOK {
+				t.Fatalf("%s: retrieve cache @%d [%d]: have %q; want %q", subj, offset, size, got, dataOK)
+			}
+		}
+
+		// retrieve [1:-1]   (also verifying that offset/size are handled correctly)
+		l := len(dataOK)
+		if l >= 2 {
+			assertCacheReadAt(1, l-2, dataOK[1:l-1])
+		}
+
+		// retrieve [:∞]
+		assertCacheReadAt(0, l+10000, dataOK)
+	}
+
+	// lookup on the file - forces to assign inode ID to it
+	// XXX better do it without kernel - how?
```

Comment:
We could register node ID to inode when it is created in `nodefs.newInode`. This is completely inside filesystem server and well defined. Now the client - the kernel - won't know that node ID before it actually tries to access corresponding file. And all cache operations on that inode should be likely noop: on retrieve it is ok to say the client has nothing, and on store most of the time it is probably ok to say ok, since the kernel is free to drop cached data. The only case is when user asked the kernel to mlock corresponding pages, but then the node ID should be already know to the kernel and thus to the handle map.

Should I modify `nID == 0` handling in store/retrieve cache according to the above logic?
Should we modify nodefs logic to automatically assign node ID right immediately when an inode is created?

If so, I think this stat can be dropped from the test.

Reply:
Please see https://github.com/hanwen/go-fuse/pull/246#discussion_r242149177. I'm dropping this XXX.